### PR TITLE
Fix geecs api scan info updating

### PR DESCRIPTION
**Pull request by skbarber**
the scan info .ini file created by MC when operating in 'remote scan' doesn't capture all of the correct metadata. We use the python api to edit it and add things like a description of the scan, scan parameter etc.

The previous method to edit this file did not use configparser, which is a cleaner way to do it. Additional string entries we're prepended to the file, but this breaks things like GEECSPlotter and the Automated scan logger. 

Note, the 'scan parameter' entry is hardcoded to be "Shotnumber". This is the way the original version worked. Seems like this would need to be changed if other kinds of scans are used via "remote scan".

### CODE REVIEW

File: GEECS-PythonAPI/geecs_python_api/controls/devices/geecs_device.py [hunk 1]
```diff
@@ -503,7 +459,56 @@ def _process_scan(cmd: str, comment: str = 'no scan', monitoring_device: Optiona
                     pass
 
         return next_folder, next_scan, accepted, timed_out
+        
+    def update_ini_file(ini_file_path, comment):
```
Comment (TAU-Reinier) [inline GEECS-PythonAPI/geecs_python_api/controls/devices/geecs_device.py:463]: Recommend marking this a `@staticmethod`. Good for clarity, but also that way geecs_device_instance.update_ini_file(path, comment) will work correctly.

**Review by TAU-Reinier — APPROVED** [review]
Good stuff. configparser is the way to go.

File: GEECS-PythonAPI/geecs_python_api/controls/devices/geecs_device.py (hunk 1 above)
Comment (skbarber) [reply GEECS-PythonAPI/geecs_python_api/controls/devices/geecs_device.py:463]: Will do. Admittedly, I’m not too knowledgeable about static vs non static

Comment (guillaumeplateau) [reply GEECS-PythonAPI/geecs_python_api/controls/devices/geecs_device.py:464]: Nice improvement. Recommend adding a check that ini_file_path is indeed a text file and not something dramatic like "C:/Program Files". I believe the "else" case would erase it all.

Comment (skbarber) [reply GEECS-PythonAPI/geecs_python_api/controls/devices/geecs_device.py:464]: I think it's safe, this part should catch any issues i think:
        # Extract number from the filename
        match = re.search(r'ScanInfoScan(\d{3})\.ini$', str(ini_file_path))
        if match:
            scan_number = int(match.group(1))
        else:
            print("Filename does not match expected format. Aborting update.")
            return

also, os.remove wouldn't delete directories either. 

I'll go ahead and merge with master and delete the branch

**Review by guillaumeplateau — APPROVED** [review]
Nice improvement

**Pull request merged**